From 2d5c09b7134dc26055a56b04657095445c854129 Mon Sep 17 00:00:00 2001
From: Aloka Dixit <quic_alokad@quicinc.com>
Date: Wed, 17 Jan 2024 10:24:48 -0800
Subject: [PATCH 1/3] wifi: ath12k: replace vendor event IDs with enum

SAWF event currently uses hardcoded value (0) for the event ID field
while calling cfg80211_vendor_event_alloc. Replace it with an enum.

Signed-off-by: Aloka Dixit <quic_alokad@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/vendor.c | 5 +++--
 drivers/net/wireless/ath/ath12k/vendor.h | 6 ++++--
 2 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/vendor.c b/drivers/net/wireless/ath/ath12k/vendor.c
index 36027267d944..baaa1d4c951e 100644
--- a/drivers/net/wireless/ath/ath12k/vendor.c
+++ b/drivers/net/wireless/ath/ath12k/vendor.c
@@ -997,7 +997,8 @@ void ath12k_vendor_telemetry_notify_breach(struct ieee80211_vif *vif, u8 *mac_ad
 		return;
 
 	skb = cfg80211_vendor_event_alloc(wdev->wiphy, wdev, NLMSG_DEFAULT_SIZE,
-					  0, GFP_KERNEL);
+					  QCA_NL80211_VENDOR_SUBCMD_SDWF_DEV_OPS_INDEX,
+					  GFP_KERNEL);
 	if (!skb) {
 		ath12k_err(NULL, "No memory available to send notify breach event\n");
 		return;
@@ -1300,7 +1301,7 @@ static struct wiphy_vendor_command ath12k_vendor_commands[] = {
 
 static const struct nl80211_vendor_cmd_info ath12k_vendor_events[] = {
 #ifdef CPTCFG_ATH12K_SAWF
-	{
+	[QCA_NL80211_VENDOR_SUBCMD_SDWF_DEV_OPS_INDEX] = {
 		.vendor_id = QCA_NL80211_VENDOR_ID,
 		.subcmd = QCA_NL80211_VENDOR_SUBCMD_SDWF_DEV_OPS,
 	},
diff --git a/drivers/net/wireless/ath/ath12k/vendor.h b/drivers/net/wireless/ath/ath12k/vendor.h
index 0d52c656d4ab..9c8fa9e6c500 100644
--- a/drivers/net/wireless/ath/ath12k/vendor.h
+++ b/drivers/net/wireless/ath/ath12k/vendor.h
@@ -11,11 +11,13 @@ enum qca_nl80211_vendor_subcmds {
 	/* Wi-Fi configuration subcommand */
 	QCA_NL80211_VENDOR_SUBCMD_SET_WIFI_CONFIGURATION = 74,
 	QCA_NL80211_VENDOR_SUBCMD_SCS_RULE_CONFIG = 218,
-#ifdef CPTCFG_ATH12K_SAWF
 	QCA_NL80211_VENDOR_SUBCMD_SDWF_PHY_OPS = 235,
 	QCA_NL80211_VENDOR_SUBCMD_SDWF_DEV_OPS = 236,
 	QCA_NL80211_VENDOR_SUBCMD_TELEMETRIC_DATA = 334,
-#endif
+};
+
+enum qca_nl80211_vendor_events {
+	QCA_NL80211_VENDOR_SUBCMD_SDWF_DEV_OPS_INDEX = 0,
 };
 
 #ifdef CPTCFG_ATH12K_SAWF
-- 
2.39.0

